packages/table/src/table.tsx

```diff
@@ -338,6 +336,50 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         selectionModes: SelectionModes.ALL,
     };
 
+    private static SHALLOWLY_COMPARABLE_PROP_KEYS = [
```

Comment:
this makes me want to be able to provide a blacklist instead of a whitelist. blacklist would contain one thing: `["selectedRegions"]`.

perhaps we should change the `keys` arg of `shallowCompareKeys` to
`{ exclude: string[] } | { include: string[] }` to accept blacklist _or_ whitelist.